client/landing/subscriptions/settings-popover/site-settings/delivery-frequency-input.tsx

```diff
@@ -0,0 +1,72 @@
+import { useTranslate } from 'i18n-calypso';
+import { useMemo } from 'react';
+import SegmentedControl from 'calypso/components/segmented-control';
+import type { SiteSubscriptionDeliveryFrequency } from '@automattic/data-stores/src/reader/types';
+
+type DeliveryFrequencyOptionProps = {
+	children: React.ReactNode;
+	value: SiteSubscriptionDeliveryFrequency;
+	selected: boolean;
+	onChange: ( value: SiteSubscriptionDeliveryFrequency ) => void;
+};
+
+const DeliveryFrequencyOption = ( {
+	children,
+	selected,
+	value,
+	onChange,
+}: DeliveryFrequencyOptionProps ) => (
+	<SegmentedControl.Item selected={ selected } onClick={ () => onChange( value ) }>
+		{ children }
+	</SegmentedControl.Item>
+);
+
+type DeliveryFrequencyInputProps = {
+	onChange: ( value: SiteSubscriptionDeliveryFrequency ) => void;
+	value: SiteSubscriptionDeliveryFrequency;
+};
+
+type DeliveryFrequencyKeyLabel = {
+	key: SiteSubscriptionDeliveryFrequency;
+	label: string;
+};
+
+const DeliveryFrequencyInput = ( {
+	onChange,
+	value: selectedValue,
+}: DeliveryFrequencyInputProps ) => {
+	const translate = useTranslate();
+	const availableFrequencies = useMemo< DeliveryFrequencyKeyLabel[] >(
+		() => [
+			{
+				key: 'instantly',
+				label: translate( 'Instantly' ),
+			},
+			{
+				key: 'daily',
+				label: translate( 'Daily' ),
+			},
+			{
+				key: 'weekly',
+				label: translate( 'Weekly' ),
+			},
+		],
+		[ translate ]
+	);
+
+	return (
+		<SegmentedControl>
+			{ availableFrequencies.map( ( { key, label } ) => (
+				<DeliveryFrequencyOption
+					selected={ selectedValue === key }
+					value={ key }
+					onChange={ onChange }
+				>
+					{ label }
+				</DeliveryFrequencyOption>
+			) ) }
```

Comment:
```suggestion
			{ availableFrequencies.map( ( { key, label }, index ) => (
				<DeliveryFrequencyOption
					selected={ selectedValue === key }
					value={ key }
					onChange={ onChange }
					key={ index }
				>
					{ label }
				</DeliveryFrequencyOption>
			) ) }
```
Adding the `key` property to the `DeliveryFrequencyOption` component removes this warning in the console:
<img width="780" alt="Screenshot 2023-04-03 at 12 53 56" src="https://user-images.githubusercontent.com/3832570/229490746-ee94f5a8-b5f2-499c-aeb7-318b815b63be.png">